File: examples/python/evm/smart_account.fund.py
# Usage: uv run python evm/smart_account.fund.py

import asyncio
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.get_or_create_account(name="account")
        smart_account = await cdp.evm.create_smart_account(owner=account)

        fund_operation = await smart_account.fund(
            network="base",
            token="usdc",
            amount=1000000,  # 1 USDC
        )

        completed_transfer = await smart_account.wait_for_fund_operation_receipt(
            transfer_id=fund_operation.id,
        )

        print(completed_transfer)


asyncio.run(main()) 

File: examples/typescript/evm/smartAccount.fund.ts
// Usage: pnpm tsx evm/smartAccount.fund.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

async function main() {
  const cdp = new CdpClient();

  const account = await cdp.evm.getOrCreateAccount({ name: "account" });
  const smartAccount = await cdp.evm.createSmartAccount({ owner: account });

  const fundOperation = await smartAccount.fund({
    network: "base",
    token: "usdc",
    amount: 1000000n, // 1 USDC
  });


  const completedTransfer = await smartAccount.waitForFundOperationReceipt({
    transferId: fundOperation.id,
  });

  console.log(completedTransfer);
}

main().catch(console.error);
